### PR TITLE
EXTPLESK-1658 publish Plesk NS records to Route53

### PR DESCRIPTION
**Pull request by maksimsavrilov**

### CODE REVIEW

File: src/plib/library/Client.php [hunk 1]
```diff
@@ -209,7 +214,7 @@ public function getHostedZoneRecordsToDelete($zoneId)
 
     public function getConfig()
     {  // Integration config
-        return [
+        $c = [
```
Comment (ktak-007) [inline src/plib/library/Client.php]: Please name your variables

File: src/plib/library/Client.php [hunk 2]
```diff
@@ -225,6 +230,10 @@ public function getConfig()
             'changeResourceRecordSets' => true,  // Permission to modify zone on AWS Route 53 free
             'deleteHostedZone' => true,  // Permission to delete zone on AWS Route 53 free
         ];
+        if(pm_Settings::get('manageNsRecords')) {
+            array_push($c['supportedTypes'],"NS");
```
Comment (ktak-007) [inline src/plib/library/Client.php]: 1. Add space after comma
2. Strings without substitution should be in single quotas.

File: src/plib/scripts/route53.php [hunk 3]
```diff
@@ -137,11 +137,18 @@
                     continue;
                 }
 
-                $uid = "CREATE {$rr->host} {$rr->type}";
+                $resourceRecordAction = "CREATE";
```
Comment (ktak-007) [inline src/plib/scripts/route53.php]: Single quotas required

File: src/plib/scripts/route53.php [hunk 4]
```diff
@@ -137,11 +137,18 @@
...
+                switch ($rr->type) {
+                    case 'NS':
+                        $resourceRecordAction = "UPSERT";
```
Comment (ktak-007) [inline src/plib/scripts/route53.php]: Single quotas required

File: src/plib/scripts/route53.php [hunk 5]
```diff
@@ -179,6 +186,29 @@
...
 
+        if(pm_Settings::get('manageNsRecords')) {
+            $nsUid = "UPSERT {$record->zone->name} NS";
+            if (array_key_exists($nsUid, $changes)) {
+                $uid = "UPSERT {$record->zone->name} SOA";
+                $soaAuthority = trim($changes[$nsUid]['ResourceRecordSet']['ResourceRecords'][0]['Value'], ' .');
+                $soaMail = trim(str_replace("@", ".", $record->zone->soa->email), ' .');
+                $soaSerial = $record->zone->soa->serial;
+                $changes[$uid] = array(
+                    'Action' => "UPSERT",
```
Comment (ktak-007) [inline src/plib/scripts/route53.php]: Single quotas required

File: src/plib/scripts/route53.php [hunk 6]
```diff
@@ -179,6 +186,29 @@
...
+                    'ResourceRecordSet' => [
+                        'Name' => $record->zone->name,
+                        'Type' => "SOA",
```
Comment (ktak-007) [inline src/plib/scripts/route53.php]: Single quotas required

File: src/plib/library/Client.php (hunk 1 above)
Comment (maksimsavrilov) [reply src/plib/library/Client.php]: fixed

**Review by ktak-007 — APPROVED** [review]

**Pull request merged**